spec/models/vote_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Vote, type: :model do
+  describe "Associations for vote on post" do
+    subject { build(:up_vote) }
+
+    it { should belong_to(:user) }
+    # it { should belong_to(:post) }
+    # error says no foreign key
```

Comment:
This is failing because shouldamatchers is testing the association, and `Vote` doesn't have an association called `:post`. You're probably looking for something like:
```ruby
it { should belong_to(:votable) }
```